packages/react-query/src/__tests__/useQuery.test.tsx

```diff
@@ -5887,4 +5887,37 @@ describe('useQuery', () => {
     fireEvent.click(fetchBtn)
     await waitFor(() => rendered.getByText('data: 3'))
   })
+
+  it('should be notified of updates between create and subscribe', async () => {
+    const key = queryKey()
+
+    function Page() {
+      const mounted = React.useRef<boolean>(false)
+      const { data, status } = useQuery({
+        enabled: false,
+        queryKey: key,
+        queryFn: async () => {
+          await sleep(10)
+          return 5
+        },
+      })
+
+      // this simulates a synchronous update between the time the query is created
```

Comment:
Please, correct me if I'm wrong, It's not important that the update was synchronous or asynchronous? You've used the synchronous update just because it's easer to test?

Reply:
Yes, exactly